### ISSUE TITLE
Deployment Failed on Preview with build success - {"code":"MissingBuildArtifacts"}

### ISSUE THREAD

**Issue by salatielosorno** [aws-amplify/amplify-hosting#3282]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### App Id

d3ly5vq7m08og3

### AWS Region

us-east-1

### Amplify Hosting feature

Frontend builds

### Describe the bug

I am facing a troubble when frontend app in deployed. Here is what Amplify logs:

```bash
2023-02-01T03:01:40 [INFO]: Beginning deployment for application d3ly5vq7m08og3, branch:development, buildId 0000000002
2023-02-01T03:01:41 [INFO]: Build Artifacts not found.
2023-02-01T03:01:41 [ERROR]: {"code":"MissingBuildArtifacts"}
```

I tried the solution given on https://github.com/aws-amplify/amplify-hosting/issues/3249 by [kennethg-singlifeph](https://github.com/kennethg-singlifeph) without success 😔

```
closing this is issue we found a resolution on this. i think bug happen on the initial build of the PR, we ecounter a please reconnect your github repository then after the rebuild deploy error happen. the solution we have was to do another pr then the issue was fix without changes on our side i think the bug happen on caching the content somewhere
```

### Expected behavior

I expect to be deployed because of I am not getting any error in Provision and Build stages.

```
.
.
.
2023-02-01T03:01:03.566Z [INFO]: info  - Compiled successfully
2023-02-01T03:01:03.566Z [INFO]: info  - Collecting page data...
2023-02-01T03:01:06.464Z [INFO]: info  - Generating static pages (0/3)
2023-02-01T03:01:06.517Z [INFO]: info  - Generating static pages (3/3)
2023-02-01T03:01:06.525Z [INFO]: info  - Finalizing page optimization...
2023-02-01T03:01:06.528Z [INFO]: 
2023-02-01T03:01:06.545Z [INFO]: Route (pages)                              Size     First Load JS
                                 ┌ ○ /                                      2.5 kB          108 kB
                                 ├   /_app                                  0 B             105 kB
                                 ├ ○ /404                                   1.47 kB         107 kB
                                 └ λ /api/hello                             0 B             105 kB
                                 + First Load JS shared by all              110 kB
                                 ├ chunks/framework-b3802df6cb251587.js   45.5 kB
                                 ├ chunks/main-a42ec09eb9be3a4c.js        29.7 kB
                                 ├ chunks/pages/_app-4c0f7851dd0ddffb.js  27.6 kB
                                 ├ chunks/webpack-33ed8e4588e25c4c.js     2.31 kB
                                 └ css/6f0e6de6ebb01b36.css               4.51 kB
2023-02-01T03:01:06.546Z [INFO]: λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
                                 ○  (Static)  automatically rendered as static HTML (uses no initial props)
2023-02-01T03:01:06.967Z [INFO]: # Completed phase: build
2023-02-01T03:01:06.970Z [INFO]: ## Build completed successfully
2023-02-01T03:01:06.970Z [INFO]: # Starting caching...
2023-02-01T03:01:06.980Z [INFO]: # Creating cache artifact...
2023-02-01T03:01:23.018Z [INFO]: # Created cache artifact
2023-02-01T03:01:23.130Z [INFO]: # Uploading cache artifact...
2023-02-01T03:01:25.813Z [INFO]: # Uploaded cache artifact
2023-02-01T03:01:25.813Z [INFO]: # Caching completed
2023-02-01T03:01:25.871Z [INFO]: # No custom headers found.
2023-02-01T03:01:25.875Z [INFO]: # Starting build artifact upload process...
2023-02-01T03:01:27.683Z [INFO]: # Uploading build artifact '__artifactsHash.zip'...
2023-02-01T03:01:27.732Z [INFO]: # Uploading build artifact '__artifacts.zip'...
2023-02-01T03:01:28.047Z [INFO]: # Build artifact upload completed
2023-02-01T03:01:28.048Z [INFO]: # Starting environment caching...
2023-02-01T03:01:28.048Z [INFO]: # Environment caching completed
Terminating logging...
```

### Reproduction steps

1.- I created the app using the CLI on account of I need to use `Web dynamic` as platform. 
> Note: I know, you will be wondering why I did not use UI to create the app. Well, I faced some problems regarding `web compute`as platform using Next.JS 12.X.X. 

2.- Connect any branch using the UI.
> Note: I try to make this using the CLI in the same way I create the app but I faced some weird problems.

### Build Settings

```yaml
version: 1
frontend:
  phases:
    preBuild:
      commands:
        - nvm install 16.15.1
        - npm install
    build:
      commands:
        - npm run build
  artifacts:
    baseDirectory: .next
    files:
      - '**/*'
  cache:
    paths:
      - node_modules/**/*
```

### Log output

<details>

```
2023-02-01T03:01:40 [INFO]: Beginning deployment for application d3ly5vq7m08og3, branch:development, buildId 0000000002
2023-02-01T03:01:41 [INFO]: Build Artifacts not found.
2023-02-01T03:01:41 [ERROR]: {"code":"MissingBuildArtifacts"}
```

</details>

### Additional information

I have the following aws and python versions installed.

```bash
aws-cli/2.1.37 Python/3.8.8
```

**Comment by qUB3r-001** [comment]
I am also facing the same issue with nx monorepo for deploying a nextjs app:

**Build settings**
```
version: 1
applications:
  - appRoot: apps/app-name
    env:
      variables:
        DUMMY: somevalue
    frontend:
      phases:
        build:
          commands:
            - cd ../../
            - npm install
            - npm list --depth=0
            - npm run nx run app-name:build:$STAGE
      artifacts:
        files:
          - '**/*'
        baseDirectory: ../../dist/apps/app-name
```

**Error output**
```
2023-02-16T12:23:41 [INFO]: Build Artifacts not found.
2023-02-16T12:23:41 [ERROR]: {"code":"MissingBuildArtifacts"}
```

**Comment by hloriana** [comment]
@qUB3r-001 can you provide your app ID as well so we can investigate why you are running into this issue?

**Comment by hloriana** [comment]
Hi @salatielosorno 👋🏽 if you are deploying a Next.js version 12 or 13 app, then you will have to use the `WEB_COMPUTE` platform. We no longer support hosting Next.js 12 or 13 applications using `WEB_DYNAMIC`. If you want to continue to use `WEB_DYNAMIC`, you will have to use Next.js version 11 or older.

> Note: I know, you will be wondering why I did not use UI to create the app. Well, I faced some problems regarding web computeas platform using Next.JS 12.X.X. 

What problems did you face with the web compute platform? We can assist you with those issues so that you can successfully host your application.

**Comment by salatielosorno** [comment]
> Hi @salatielosorno 👋🏽 if you are deploying a Next.js version 12 or 13 app, then you will have to use the `WEB_COMPUTE` platform. We no longer support hosting Next.js 12 or 13 applications using `WEB_DYNAMIC`. If you want to continue to use `WEB_DYNAMIC`, you will have to use Next.js version 11 or older.
> 
> > Note: I know, you will be wondering why I did not use UI to create the app. Well, I faced some problems regarding web computeas platform using Next.JS 12.X.X.
> 
> What problems did you face with the web compute platform? We can assist you with those issues so that you can successfully host your application.

My applications keep blank, app just opened with the background color but no elements are shown. New App Id `d302rov4yt189u`

**Comment by qUB3r-001** [comment]
> @qUB3r-001 can you provide your app ID as well so we can investigate why you are running into this issue?

Sorry for the late reply @hloriana . After trying to make it work for hours i ended up deleteing the amplify app and for now working with poly-repo flow with indivisual nextjs app in each repo.
Although i would like to know about some things:

1. When using nx generator to build nextjs app `nrwl/next:build` it creates the build package in the path `dist/my-app` which contains `.next` and other parts as well. Whereas normally nextjs build creates only `.next` which is used to deploy. So for artifacts what should i use in the case when deploying with `nx` ?
2. One of the issue while using amplify monorepo settings is whenever there is a commit in the linked branch the build would be triggered for all the associated amplify apps ( Ideally i would like to only build app which has changed or `affected` in nx terms). 
To achieve this i was thinking of making release branches for each app and just `nx run build:<app-name-of-that-branch>`. So is this a good way to go or what should i do to make this monorepo deployment flow work with amplify?

TLDR; **Have a monorepo for multiple nextjs apps and deploy them via amplify and re-deploy a particular app only when it changes and not just on any change in the monorepo**

**Comment by michrome** [comment]
_Noting it addresses only 1 [above](https://github.com/aws-amplify/amplify-hosting/issues/3282#issuecomment-1449951964)_—Now available as a beta, Amplify Hosting has improved support for monorepo frameworks including `npm/yarn/pnpm workspaces`, `Turborepo` and `Nx`. If you'd like to try it out, please let me know your AWS Account ID number so we can flag you in. My email is `jrm@amazon.com`.

**Comment by salatielosorno** [comment]
Suddenly without any reason it works. I have just reconnected the repository some minutes ago and all is working well. I think since I reported the issue it has been some changes in Amplify that fix it.
